src/wasm/wasm-ir-builder.cpp

```diff
@@ -1441,17 +1441,29 @@ Result<> IRBuilder::makePop(Type type) {
   // already create them automatically when starting a legacy catch block that
   // needs one. Just verify that the Pop we are being asked to make is the same
   // type as the Pop we have already made.
-  auto& scope = getScope();
-  if (!scope.getCatch() || scope.exprStack.size() != 1 ||
-      !scope.exprStack[0]->is<Pop>()) {
-    return Err{
-      "pop instructions may only appear at the beginning of catch blocks"};
-  }
-  auto expectedType = scope.exprStack[0]->type;
-  if (!Type::isSubType(expectedType, type)) {
-    return Err{std::string("Expected pop of type ") + expectedType.toString()};
+  //
+  // We need to look up the scope stack, as we can look through nameless blocks.
+  Index i = 0;
+  while (1) {
+    auto scopeCheck = getScope(i++);
+    CHECK_ERR(scopeCheck);
+    auto& scope = **scopeCheck;
+    if (auto* block = scope.getBlock()) {
+      if (!block->name.is()) {
+        continue;
+      }
+    }
+    if (!scope.getCatch() || scope.exprStack.size() != 1 ||
+        !scope.exprStack[0]->is<Pop>()) {
+      return Err{
+        "pop instructions may only appear at the beginning of catch blocks"};
+    }
+    auto expectedType = scope.exprStack[0]->type;
+    if (!Type::isSubType(expectedType, type)) {
+      return Err{std::string("Expected pop of type ") + expectedType.toString()};
+    }
+    return Ok{};
```

Comment:
This change avoids the error, but it doesn't actually move the pop into the right place. Whenever we try to pop from an empty stack, we should look up the stack through nameless blocks to find pops that we could use.

Weird edge case to test:

 - We lazily give unnamed blocks names when we encounter branches that target them via an index. What happens if we have previously tried to pull a `pop` through such a block?

Reply:
Why is this a situation of popping from an empty stack? Are `Pop` instructions not added to the stack like anything else? (they have a concrete type)

Reply:
No, `pop`s are special. They are automatically pushed onto the stack when the beginning of a `catch` is parsed, not when the `pop` instruction itself is parsed. That's why the existing code here just does some error checking and doesn't push anything. The reason is that we fully support parsing standard code that does not contain any `pop`s, so we can't wait until we see a `pop` to push it onto the stack. For example, this parses:

```
try
catch $e
 drop
```